src/openassetio-python/package/openassetio/pluginSystem/PythonPluginSystemManagerImplementationFactory.py

```diff
@@ -37,48 +37,97 @@ class PythonPluginSystemManagerImplementationFactory(ManagerImplementationFactor
     kPluginEnvVar = "OPENASSETIO_PLUGIN_PATH"
+    ## The Environment Variable to control the discovery of entry point based plugins
+    kDisableEntryPointsEnvVar = "OPENASSETIO_DISABLE_ENTRYPOINTS_PLUGINS"
+
+    ## The name of the ManagerPlugin entry point for entry point
+    ## discovered plugins.
+    kPackageEntryPointName = "openassetio.manager_plugin"
+
+    def __init__(self, logger, paths=None, disableEntryPointsPlugins=None):
+        """
+        Creates a new factory. The factory scans for plugins lazily on
+        the first invocation of @ref identifiers or @ref instantiate.
 
-    def __init__(self, logger, paths=None):
+        @param logger @fqref{log.LoggerInterface} "LoggerInterface"
+        implementation that will be used to output information about
+        plugin loading.
+
+        @param paths `str` Paths to search for plugins. Defaults to the
+        value of the @ref kPluginEnvVar environment variable.
+
+        @param disableEntryPointsPlugins `bool` Controls whether
+        package entry point based plugin discovery is allowed. Defaults
+        to False unless the @ref kDisableEntryPointsEnvVar environment
+        variable is set.
+        """
 
         super(PythonPluginSystemManagerImplementationFactory, self).__init__(logger)
 
         self.__pluginManager = None
         self.__paths = paths
+        self.__disableEntryPointsPlugins = disableEntryPointsPlugins
 
     def __scan(self):
         """
         Scans for PythonPluginSystemManagerPlugins, and registers them
         with the factory instance.
-
-        @param paths `str` A searchpath string to search for plug-ins.
-        If None, then the contents of the Environment Variable
-        @ref kPluginEnvVar is used instead.
         """
 
-        if not self.__paths:
+        if self.__disableEntryPointsPlugins is None:
+            self.__disableEntryPointsPlugins = os.environ.get(
+                self.kDisableEntryPointsEnvVar, False
+            )
+
+        if self.__paths is None:
             self.__paths = os.environ.get(self.kPluginEnvVar, "")
-            if not self.__paths:
-                self._logger.log(
-                    self._logger.Severity.kWarning,
-                    "%s is not set. It is somewhat unlikely that you will find any plugins..."
-                    % self.kPluginEnvVar,
-                )
 
+        # Construct this here, so we have this even if we early out
         self.__pluginManager = PythonPluginSystem(self._logger)
 
-        # We do this after instantiating, so that the lifetime of the manager is
-        # consistent with cases where some paths were set.
+        if not self.__paths and self.__disableEntryPointsPlugins:
+            self._logger.log(
+                self._logger.Severity.kWarning,
+                "No search paths specified and entry point plugins are disabled, no plugins "
+                f"will load - check ${self.kPluginEnvVar} is set.",
```

Comment:
:+1:  Love this. Avoid people bashing their heads against silent terminals wondering why nothing is working.